lib/tasks/data/orphaned_records_queries.sql

```diff
@@ -0,0 +1,209 @@
+SELECT
+	COUNT(*)
+FROM
+	spree_stock_movements
+LEFT JOIN spree_stock_items on
+	spree_stock_movements.stock_item_id = spree_stock_items.id
+WHERE
+	spree_stock_items.id IS NULL
+
+SELECT
+	COUNT(*)
+FROM
+	spree_stock_locations
+LEFT JOIN spree_states on
+	spree_stock_locations.state_id = spree_states.id
+WHERE
+	spree_states.id IS NULL
+
+SELECT
+	COUNT(*)
+FROM
+	spree_stock_locations
+LEFT JOIN spree_countries on
+	spree_stock_locations.country_id = spree_countries.id
+WHERE
+	spree_countries.id IS NULL
+	AND spree_stock_locations.state_id IS NOT NULL
```

Comment:
So you don't have code to generate this any more, do you? And I noticed that the state id is still assumed to be mandatory. I think for the purpose of adding foreign keys, we can add the check for optional id (NULL) to every query.

Could the queries be added to the migrations? They could even be in a comment just to advise about a pre-check. Then the query and the migration are in the same place and it's easier to keep in sync.